src/Datadog.NativeLoader/proxy.cpp

```diff
@@ -0,0 +1,394 @@
+        if (m_canUnloadNow != nullptr)
+        {
+            return m_canUnloadNow();
+        }
+
+        // The function cannot be loaded.
+        return E_FAIL;
+    }
+
+    ICorProfilerCallback10* DynamicInstance::GetProfilerCallback()
+    {
+        return m_corProfilerCallback;
+    }
+
+    std::string DynamicInstance::GetFilePath()
+    {
+        return m_filepath;
+    }
+
+    std::string DynamicInstance::GetClsId()
+    {
+        return m_clsid_string;
+    }
+
+    // ************************************************************************
+
+    //
+    // public
+    //
+
+    DynamicDispatcher::DynamicDispatcher()
+    {
+        m_instances = std::vector<std::unique_ptr<DynamicInstance>>();
+    }
+
+    void DynamicDispatcher::Add(std::unique_ptr<DynamicInstance>& instance)
+    {
+        m_instances.push_back(std::move(instance));
+    }
+
+    void DynamicDispatcher::LoadConfiguration(std::string configFilePath)
+    {
+        if (!std::filesystem::exists(configFilePath))
+        {
+            Warn("Configuration file doesn't exist.");
+            return;
+        }
+
+        std::unordered_map<std::string, bool> guidBoolMap;
+        std::ifstream t;
+        t.open(configFilePath);
+
+        // Gets the configuration file folder
+        std::filesystem::path configFolder = std::filesystem::path(configFilePath).remove_filename();
+
+        // Get the current path
+        std::filesystem::path oldCurrentPath = std::filesystem::current_path();
+
+        // Set the current path to the configuration folder (to allow relative paths)
+        std::filesystem::current_path(configFolder);
+
+        while (t)
+        {
+            std::string line;
+            std::getline(t, line);
+            line = Trim(line);
+            if (line.length() != 0)
+            {
+                Debug(line);
+
+                if (line[0] == '#')
+                {
+                    continue;
+                }
+
+                std::vector<std::string> lineArray = Split(line, ';');
+                std::string idValue = lineArray[0];
+                std::string osArchValue = lineArray[1];
+                std::string filepathValue = lineArray[2];
+
+                if (osArchValue == currentOsArch)
+                {
+                    // Convert possible relative paths to absolute paths using the configuration file folder as base
+                    // (current_path)
+                    filepathValue = std::filesystem::absolute(filepathValue).string();
+                    if (std::filesystem::exists(filepathValue))
+                    {
+                        guidBoolMap[idValue] = true;
+                        std::unique_ptr<DynamicInstance> instance =
+                            std::make_unique<DynamicInstance>(filepathValue, idValue);
+                        this->Add(instance);
+                        WSTRING env_key = WStr("PROFID_") + ToWSTRING(idValue);
```

Comment:
Because the tracer is using `CORECLR_PROFILER_PATH[_XX]` on non Windows OS's to set the `PInvokeMap` target lib (absolute path) in the loader when we add `GetAssemblyAndSymbolsBytes` extern [here](https://github.com/DataDog/dd-trace-dotnet/blob/tony/native-loader/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp#L2218). 
And due the tracer corprofiler filepath is not longer in that env-var, it tries to load that `GetAssemblyAndSymbolsBytes` function from the native loader, failing because that symbol doesn't exist.

The workaround in this case is to set a new environment variable with the format: `PROFID_[COR_IID] = [filepath]`. This workaround also requires the change in the tracer to use that environment variable before trying other one [here](https://github.com/DataDog/dd-trace-dotnet/blob/tony/native-loader/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp#L2212) .